Composer/packages/client/src/pages/design/index.tsx

```diff
@@ -425,28 +430,43 @@ function DesignPage(props) {
           </Conversation>
         </div>
       </div>
-      {state.showCreateDialogModal && (
-        <CreateDialogModal
-          isOpen={state.showCreateDialogModal}
-          onDismiss={() => actions.createDialogCancel()}
-          onSubmit={handleCreateDialogSubmit}
-        />
-      )}
-      {state.showAddSkillDialogModal && (
-        <AddSkillDialog
-          isOpen={state.showAddSkillDialogModal}
-          onDismiss={() => actions.addSkillDialogCancel()}
-          onSubmit={handleAddSkillDialogSubmit}
-        />
-      )}
-      {triggerModalVisible && (
-        <TriggerCreationModal
-          dialogId={dialogId}
-          isOpen={triggerModalVisible}
-          onDismiss={onTriggerCreationDismiss}
-          onSubmit={onTriggerCreationSubmit}
-        />
-      )}
+      <Suspense fallback={<div />}>
+        {state.showCreateDialogModal && (
+          <CreateDialogModal
+            isOpen={state.showCreateDialogModal}
+            onDismiss={() => actions.createDialogCancel()}
+            onSubmit={handleCreateDialogSubmit}
+          />
+        )}
+      </Suspense>
+      <Suspense fallback={<div />}>
+        {state.showAddSkillDialogModal && (
+          <AddSkillDialog
+            isOpen={state.showAddSkillDialogModal}
+            onDismiss={() => actions.addSkillDialogCancel()}
+            onSubmit={handleAddSkillDialogSubmit}
+          />
+        )}
+      </Suspense>
+      <Suspense fallback={<div />}>
+        {exportSkillModalVisible && (
+          <ExportSkillModal
+            isOpen={exportSkillModalVisible}
+            onDismiss={() => setExportSkillModalVisible(false)}
+            onSubmit={() => setExportSkillModalVisible(false)}
+          />
+        )}
+      </Suspense>
+      <Suspense fallback={<div />}>
+        {triggerModalVisible && (
+          <TriggerCreationModal
+            dialogId={dialogId}
+            isOpen={triggerModalVisible}
+            onDismiss={onTriggerCreationDismiss}
+            onSubmit={onTriggerCreationSubmit}
+          />
+        )}
+      </Suspense>
```

Comment:
This just needs 1 Suspense wrapper. Also consider using the <LoadingSpinner /> component.